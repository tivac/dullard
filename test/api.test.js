"use strict";

var tests   = require("./lib/tests.js"),
    Dullard = require("../src/dullard.js");

describe("Dullard", function() {
    describe("API", function() {
        describe(".clone()", function() {
            it("should return a new instance", () => {
                var d = new Dullard(),
                    c;

                d.addConfig({
                    one : 1,
                    steps : {
                        one() {},
                        default : [
                            () => {}
                        ]
                    }
                });

                c = d.clone();

                d.one = 2;

                expect(c).toMatchSnapshot();
                expect(d).toMatchSnapshot();
            });
        });

        describe(".child", function() {
            it("should run in a clone of the main instance", () => {
                var d = new Dullard(),
                    c;
                
                d.addConfig({
                    steps : {
                        one() { }
                    }
                });

                return d.child("one")
                    .then((c) => {
                        d.one = 2;

                        expect(c).toMatchSnapshot();
                        expect(d).toMatchSnapshot();
                    });
            });

            it("should return expected errors", () => {
                var d = new Dullard(),
                    c;
                
                d.addConfig({
                    steps : {
                        one() { }
                    }
                });

                return d.child(
                    "two"
                )
                .catch((err) => {
                    expect(err).toMatchSnapshot();
                });
            });
        });
    });
});
