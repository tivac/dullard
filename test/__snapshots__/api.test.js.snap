// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dullard API .child should return expected errors 1`] = `[Error: Unknown task: two]`;

exports[`Dullard API .child should run in a clone of the main instance 1`] = `
Dullard {
  "_current": null,
  "_events": Object {},
  "_eventsCount": 0,
  "_maxListeners": undefined,
  "config": Object {
    "cwd": "D:\\\\git\\\\dullard",
    "dirs": Array [],
    "dullard": [Circular],
    "files": Array [],
    "log": [Function],
    "steps": Object {
      "one": [Function],
    },
  },
  "domain": null,
  "steps": Object {
    "one": [Function],
  },
  "tasks": Object {
    "one() {}": [Function],
  },
  "testing": false,
}
`;

exports[`Dullard API .child should run in a clone of the main instance 2`] = `
Dullard {
  "_events": Object {},
  "_eventsCount": 0,
  "_maxListeners": undefined,
  "config": Object {
    "cwd": "D:\\\\git\\\\dullard",
    "dirs": Array [],
    "dullard": [Circular],
    "files": Array [],
    "log": [Function],
    "steps": Object {
      "one": [Function],
    },
  },
  "domain": null,
  "one": 2,
  "steps": Object {
    "one": [Function],
  },
  "tasks": Object {},
  "testing": false,
}
`;

exports[`Dullard API .clone() should return a new instance 1`] = `
Dullard {
  "_events": Object {},
  "_eventsCount": 0,
  "_maxListeners": undefined,
  "config": Object {
    "cwd": "D:\\\\git\\\\dullard",
    "dirs": Array [],
    "dullard": [Circular],
    "files": Array [],
    "log": [Function],
    "one": 1,
    "steps": Object {
      "default": Array [
        [Function],
      ],
      "one": [Function],
    },
  },
  "domain": null,
  "steps": Object {
    "default": Array [
      [Function],
    ],
    "one": [Function],
  },
  "tasks": Object {},
  "testing": false,
}
`;

exports[`Dullard API .clone() should return a new instance 2`] = `
Dullard {
  "_events": Object {},
  "_eventsCount": 0,
  "_maxListeners": undefined,
  "config": Object {
    "cwd": "D:\\\\git\\\\dullard",
    "dirs": Array [],
    "dullard": [Circular],
    "files": Array [],
    "log": [Function],
    "one": 1,
    "steps": Object {
      "default": Array [
        [Function],
      ],
      "one": [Function],
    },
  },
  "domain": null,
  "one": 2,
  "steps": Object {
    "default": Array [
      [Function],
    ],
    "one": [Function],
  },
  "tasks": Object {},
  "testing": false,
}
`;
